### PR TITLE
API Editor can preview unpublished articles SHOW

### PR DESCRIPTION
**Pull request by kermit-klein**
[PT LINK](https://www.pivotaltracker.com/story/show/173037242)

- Add Article image in Factory Bot
- Add show action to admin article controller
- Add show action serializer for admin article controller
- Add request spec with sad and happy paths

### CODE REVIEW

**Review by ViaMarcus — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -135,6 +136,31 @@ Bad params, example:
...
+to editors only
+
+```
+"article": 
+  {
+    "id":1,
+    "title":"title1",
+    "category":"category1",
+    "created_at":"YYYY-MM-dd hh:mm"
+  }
```
Comment (ViaMarcus) [inline README.md:153]: And image?

Comment (ViaMarcus) [reply README.md:153]: And body?

File: app/controllers/api/admin/articles_controller.rb [hunk 2]
```diff
@@ -5,8 +5,19 @@ class Api::Admin::ArticlesController < ApplicationController
...
-      article = Article.where(published: false)
-      render json: article, each_serializer: Admin::Article::IndexSerializer
+    article = Article.where(published: false)
+    render json: article, each_serializer: Admin::Article::IndexSerializer
+  end
+
+  def show
+    article = Article.find(params[:id])
+    if article.published == true
+      render json: { message: 'This article was already published' }, status: 400
```
Comment (jysmys) [inline app/controllers/api/admin/articles_controller.rb]: Http 400 response is a bit weird for this, think 422 is better....

File: spec/factories/articles.rb [hunk 3]
```diff
@@ -5,5 +5,8 @@
     title { 'this is title' }
     body { 'this is body, lorem ipsum.' * 20 }
     category { 'sport' }
+    trait :with_image do
```
Comment (jysmys) [inline spec/factories/articles.rb:8]: what is trait?

**Review by jysmys — APPROVED** [review]

**Review by erikbjoern — APPROVED** [review]

File: spec/factories/articles.rb (hunk 3 above)
Comment (erikbjoern) [reply spec/factories/articles.rb:8]: I think that's what you have to write when it `has_one_attached`.

File: app/controllers/api/admin/articles_controller.rb (hunk 2 above)
Comment (erikbjoern) [reply app/controllers/api/admin/articles_controller.rb]: Hm... both works but I think 400, 'bad request' is more suiting. You're not supposed to ask for published articles here.

File: README.md (hunk 1 above)
Comment (erikbjoern) [reply README.md:153]: Wups... will soon be updated.

File: app/controllers/api/admin/articles_controller.rb (hunk 2 above)
Comment (erikbjoern) [reply app/controllers/api/admin/articles_controller.rb]: Well, okay, now that I think of it, I think you're right. Bad request is more about bad syntax I guess, and unprocessable entity can very well be good syntax but not something that we will process.

File: spec/factories/articles.rb (hunk 3 above)
Comment (kermit-klein) [reply spec/factories/articles.rb:8]: Since it is not a db column, it needs some special care
Polymorphic associations can be handled with traits

**Review by designerofthing — APPROVED** [review]

**Review by PaulineBA — APPROVED** [review]

**Review by PaulineBA — APPROVED** [review]

File: app/controllers/api/admin/articles_controller.rb [hunk 4]
```diff
@@ -5,8 +5,19 @@ class Api::Admin::ArticlesController < ApplicationController
...
+  end
+
+  def show
+    article = Article.find(params[:id])
+    if article.published == true
+      render json: { message: 'This article was already published' }, status: 422
+    else
+      render json: article, serializer: Admin::Article::ShowSerializer
+    end
+  rescue StandardError => e
```
Comment (oliverochman) [inline app/controllers/api/admin/articles_controller.rb]: this here you can extract with 

`rescue_from ActiveRecord::RecordNotFound, with: :render_active_record_error`

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/api/admin/articles_controller.rb (hunk 4 above)
Comment (kermit-klein) [reply app/controllers/api/admin/articles_controller.rb]: How do i use it ?

**Review by oliverochman — APPROVED** [review]

**Pull request merged**